### PR TITLE
Support for ios releated commands and iTunes x64

### PR DESCRIPTION
**Pull request by Fatme**
http://teampulse.telerik.com/view#item/285187
http://teampulse.telerik.com/view#item/285576

### CODE REVIEW

File: mobile/ios/ios-core.ts [hunk 1]
```diff
@@ -109,7 +109,7 @@ class IOSCore implements Mobile.IiOSCore {
 	}
 
 	private get CommonProgramFilesPath(): string {
-		return hostInfo.isWindows64() ?  process.env["CommonProgramFiles(x86)"] : process.env.CommonProgramFiles;
+		return process.arch === "x64" ? process.env.CommonProgramFiles : hostInfo.isWindows32() ? process.env["CommonProgramFiles(x86)"] : process.env.CommonProgramFiles;
```
Comment (teobugslayer) [inline mobile/ios/ios-core.ts]: I believe that ```return process.env.CommonProgramFiles``` works for both 32 and 64 bit code

File: validators/iTunes-validator.ts [hunk 2]
```diff
@@ -7,39 +7,68 @@ import hostInfo = require("../host-info");
...
+		return this.isiTunesInstalledCore(coreFoundationDir, mobileDeviceDir);
+	}
+
+	private isiTunesInstalledCore(coreFoundationDir: string, mobileDeviceDir: string): IFuture<boolean> {
+		return (() => {
+			if(!this.$fs.exists(coreFoundationDir).wait() || !this.$fs.exists(mobileDeviceDir).wait()) {
+				return false;
+			}
+
+			return true;
```
Comment (teobugslayer) [inline validators/iTunes-validator.ts]: ``` return this.$fs.exists(coreFoundationDir).wait() && this.$fs.exists(mobileDeviceDir).wait();```

File: validators/iTunes-validator.ts [hunk 3]
```diff
@@ -7,39 +7,68 @@ import hostInfo = require("../host-info");
...
-			var coreFoundationDir = "";
-			var mobileDeviceDir = "";
+				if(isNode64) { //x64-windows
+					commonProgramFiles = process.env.CommonProgramFiles;
+					if(this.isiTunesInstalledOnWindows(process.env["CommonProgramFiles(x86)"]).wait() && !this.isiTunesInstalledOnWindows(commonProgramFiles).wait()) {
+						return ITunesValidator.BITNESS_MISMATCH_ERROR_MESSAGE;
+					}
+				} else {
+					if(hostInfo.isWindows32()) { // x86-node, x86-windows
+						return process.env.CommonProgramFiles;
```
Comment (teobugslayer) [inline validators/iTunes-validator.ts]: Should this be returned at all?

**Comment by teobugslayer** [comment]
Excellent work! :+1: and then :100: and :1234:

**Comment by rosen-vladimirov** [comment]
:+1:

File: validators/iTunes-validator.ts [hunk 4]
```diff
@@ -7,39 +7,63 @@ import hostInfo = require("../host-info");
 
 export class ITunesValidator implements Mobile.IiTunesValidator {
 	private static NOT_INSTALLED_iTUNES_ERROR_MESSAGE = "iTunes is not installed. Install it on your system and run this command again.";
+	private static BITNESS_MISMATCH_ERROR_MESSAGE = "The bitness of Node.js and iTunes must match. Verify that both Node.js and iTunes are 32-bit or 64-bit and try again.";
```
Comment (tytodorov) [inline validators/iTunes-validator.ts]: WE can use: Verify that both Node.js and iTunes are respectively 32-bit or 64-bit and try again."

**Comment by tytodorov** [comment]
:+1: The above comment is just a suggestion.

**Pull request merged**